jwst/extract_1d/soss_extract/pastasoss.py

```diff
@@ -0,0 +1,537 @@
+        wavelengths = get_wavelengths(refmodel, x_new, pwcpos, int(order))
+
+        # return x_new, y_new, wavelengths
+        return order, x_new, y_new, wavelengths
+
+    else:
+        error_message = f"Order {order} is not supported at this time."
+        log.error(error_message)
+        raise ValueError(error_message)
+
+
+def extrapolate_to_wavegrid(w_grid, wavelength, quantity):
+    """
+    Extrapolates quantities on the right and the left of a given array of quantity
+
+    Parameters
+    ----------
+    w_grid : sequence
+        The wavelength grid to interpolate onto
+    wavelength : sequence
+        The native wavelength values of the data
+    quantity : sequence
+        The data to interpolate
+
+    Returns
+    -------
+    Array
+        The interpolated quantities
+    """
+    sorted = np.argsort(wavelength)
+    q = quantity[sorted]
+    w = wavelength[sorted]
+
+    # Determine the slope on the right of the array
+    slope_right = (q[-1] - q[-2]) / (w[-1] - w[-2])
+    # extrapolate at wavelengths larger than the max on the right
+    indright = np.where(w_grid > w[-1])[0]
+    q_right = q[-1] + (w_grid[indright] - w[-1]) * slope_right
+    # Determine the slope on the left of the array
+    slope_left = (q[1] - q[0]) / (w[1] - w[0])
+    # extrapolate at wavelengths smaller than the min on the left
+    indleft = np.where(w_grid < w[0])[0]
+    q_left = q[0] + (w_grid[indleft] - w[0]) * slope_left
+    # Construct and extrapolated array of the quantity
+    w = np.concatenate((w_grid[indleft], w, w_grid[indright]))
+    q = np.concatenate((q_left, q, q_right))
+
+    # resample at the w_grid everywhere
+    q_grid = np.interp(w_grid, w, q)
+
+    return q_grid
+
+
+def calc_2d_wave_map(wave_grid, x_dms, y_dms, tilt, oversample=2, padding=10, maxiter=5, dtol=1e-2):
+    """Compute the 2D wavelength map on the detector.
+
+    Parameters
+    ----------
+    wave_grid : sequence
+        The wavelength corresponding to the x_dms, y_dms, and tilt values.
+    x_dms : sequence
+        The trace x position on the detector in DMS coordinates.
+    y_dms : sequence
+        The trace y position on the detector in DMS coordinates.
+    tilt : sequence
+        The trace tilt angle in degrees.
+    oversample : int
+        The oversampling factor of the input coordinates.
+    padding : int
+        The native pixel padding around the edge of the detector.
+    maxiter : int
+        The maximum number of iterations used when solving for the wavelength at each pixel.
+    dtol : float
+        The tolerance of the iterative solution in pixels.
+
+    Returns
+    -------
+    Array
+        An array containing the wavelength at each pixel on the detector.
+    """
+    os = np.copy(oversample)
+    xpad = np.copy(padding)
+    ypad = np.copy(padding)
+
+    # No need to compute wavelengths across the entire detector, slightly larger than SUBSTRIP256 will do.
+    dimx, dimy = 2048, 300
```

Comment:
consider moving hard-coded values into all-caps variable names at top of file

Reply:
Did my best to move all hard-coded values to const vars.